### PR TITLE
C++: Fix more FPs with extraction errors on cpp/wrong-type-format-arguments

### PR DESCRIPTION
**Pull request by calumgrant**
#### All query authors

- [ ] A change note is added if necessary. See [the documentation](https://github.com/github/codeql/blob/main/docs/change-notes.md) in this repository.
- [ ] All new queries have appropriate `.qhelp`. See [the documentation](https://github.com/github/codeql/blob/main/docs/query-help-style-guide.md) in this repository.
- [ ] QL tests are added if necessary. See [Testing custom queries](https://docs.github.com/en/code-security/codeql-cli/using-the-advanced-functionality-of-the-codeql-cli/testing-custom-queries) in the GitHub documentation.
- [ ] New and changed queries have correct query metadata. See [the documentation](https://github.com/github/codeql/blob/main/docs/query-metadata-style-guide.md) in this repository.

#### Internal query authors only

- [ ] Autofixes generated based on these changes are valid, only needed if this PR makes significant changes to `.ql`, `.qll`, or `.qhelp` files. See [the documentation](https://github.com/github/codeql-team/blob/main/docs/best-practices/validating-autofix-for-query-changes.md) (internal access required).
- [ ] Changes are validated [at scale](https://github.com/github/codeql-dca/) (internal access required).
- [ ] Adding a new query? Consider also [adding the query to autofix](https://github.com/github/codeml-autofix/blob/main/docs/updating-query-support.md#adding-a-new-query-to-the-query-suite).

### CODE REVIEW

File: cpp/ql/src/Likely Bugs/Format/WrongTypeFormatArguments.ql [hunk 1]
```diff
@@ -171,6 +171,10 @@ where
   not arg.isAffectedByMacro() and
   not arg.isFromUninstantiatedTemplate(_) and
   not actual.getUnspecifiedType() instanceof ErroneousType and
+  not (
+    expected instanceof PointerType and
+    actual.getUnspecifiedType().(PointerType).getBaseType() instanceof ErroneousType
+  ) and
```
Comment (jketema) [inline cpp/ql/src/Likely Bugs/Format/WrongTypeFormatArguments.ql:177]:
```suggestion
  not actual.stripType() instanceof ErroneousType and
```

**Comment by jketema** [comment]
Would you mind re-running DCA, as the introduction of `stripType` is a rather fundamental change?

**Review by jketema — COMMENTED** [review]
LGMT, just one fix for the change note.

**Comment by calumgrant** [comment]
Are you ok with the DCA build failure on lubomyr__bochs? It looks unrelated.

**Review by jketema — APPROVED** [review]

**Comment by jketema** [comment]
> Are you ok with the DCA build failure on lubomyr__bochs? It looks unrelated.

Yes. It's a tracer injection failure. We have an open issue for it somewhere.

**Pull request merged**